File: src/main/java/cope/nebula/client/feature/command/impl/Toggle.java
package cope.nebula.client.feature.command.impl;

import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import com.mojang.brigadier.builder.RequiredArgumentBuilder;
import com.mojang.realmsclient.gui.ChatFormatting;
import cope.nebula.client.feature.command.Command;
import cope.nebula.client.feature.command.type.ModuleArgumentType;
import cope.nebula.client.feature.module.Module;

public class Toggle extends Command {
    public Toggle() {
        super("toggle", LiteralArgumentBuilder.literal("toggle")
                .then(RequiredArgumentBuilder.argument("module", ModuleArgumentType.module())
                        .executes((ctx) -> {
                            Module module = ModuleArgumentType.getModule(ctx, "module");
                            module.toggle();

                            send(module.getName() + " toggled " + (
                                    module.isOn() ?
                                            ChatFormatting.GREEN + "on" + ChatFormatting.RESET :
                                            ChatFormatting.RED + "off" + ChatFormatting.RESET) + ".",
                                    module.hashCode());

                            return 0;
                        }))
                .executes((ctx) -> {
                    send("Please provide a module name.");
                    return 0;
                }));
    }
}

File: src/main/kotlin/wtf/nebula/client/feature/command/impl/Toggle.kt
package wtf.nebula.client.feature.command.impl

import com.mojang.brigadier.Command.SINGLE_SUCCESS
import com.mojang.brigadier.arguments.StringArgumentType
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import com.mojang.brigadier.builder.RequiredArgumentBuilder
import wtf.nebula.client.feature.command.Command
import wtf.nebula.client.registry.impl.CommandRegistry.Companion.INVALID_USAGE
import wtf.nebula.client.registry.impl.ModuleRegistry

class Toggle : Command(arrayOf("toggle", "t", "on", "off"), "[module]", "Toggles a module") {
    override fun build(builder: LiteralArgumentBuilder<Any>) {
        builder.then(
            RequiredArgumentBuilder.argument<Any, String>("module", StringArgumentType.string()).executes {
                val moduleName = StringArgumentType.getString(it, "module")
                if (moduleName.isNullOrEmpty()) {
                    return@executes send("Invalid module name")
                }

                val module = ModuleRegistry.INSTANCE!!.registers.find {
                        mod -> mod.name.equals(moduleName, ignoreCase = true)
                } ?: return@executes send("Invalid module name")

                module.toggled = !module.toggled
                return@executes SINGLE_SUCCESS
            }
        ).executes {
            return@executes INVALID_USAGE
        }
    }
}